extractor/src/test/java/org/schabi/newpipe/extractor/services/DefaultStreamExtractorTest.java

```diff
@@ -0,0 +1,382 @@
+    }
+
+    @Test
+    @Override
+    public void testLikeCount() throws Exception {
+        if (expectedLikeCountAtLeast() == -1) {
+            assertEquals(-1, extractor().getLikeCount());
+        } else {
+            assertAtLeast(expectedLikeCountAtLeast(), extractor().getLikeCount());
+        }
+    }
+
+    @Test
+    @Override
+    public void testDislikeCount() throws Exception {
+        if (expectedDislikeCountAtLeast() == -1) {
+            assertEquals(-1, extractor().getDislikeCount());
+        } else {
+            assertAtLeast(expectedDislikeCountAtLeast(), extractor().getDislikeCount());
+        }
+    }
+
+    @Test
+    @Override
+    public void testRelatedStreams() throws Exception {
+        final StreamInfoItemsCollector relatedStreams = extractor().getRelatedStreams();
+
+        if (expectedHasRelatedStreams()) {
+            defaultTestListOfItems(extractor().getService(), relatedStreams.getItems(),
+                    relatedStreams.getErrors());
+        } else {
+            assertNull(relatedStreams);
+        }
+    }
+
+    @Test
+    @Override
+    public void testAgeLimit() throws Exception {
+        assertEquals(expectedAgeLimit(), extractor().getAgeLimit());
+    }
+
+    @Test
+    @Override
+    public void testErrorMessage() throws Exception {
+        assertEquals(expectedErrorMessage(), extractor().getErrorMessage());
+    }
+
+    @Test
+    @Override
+    public void testVideoStreams() throws Exception {
+        final List<VideoStream> videoStreams = extractor().getVideoStreams();
+        final List<VideoStream> videoOnlyStreams = extractor().getVideoOnlyStreams();
+        assertNotNull(videoStreams);
+        assertNotNull(videoOnlyStreams);
+        videoStreams.addAll(videoOnlyStreams);
+
+        if (expectedHasVideoStreams()) {
+            assertFalse(videoStreams.isEmpty());
+
+            for (final VideoStream stream : videoStreams) {
+                assertIsSecureUrl(stream.getUrl());
+                assertFalse(stream.getResolution().isEmpty());
+
+                final int formatId = stream.getFormatId();
+                assertTrue("format id does not fit a video stream: " + formatId,
+                        0 <= formatId && formatId < 0x100);
+            }
+        } else {
+            assertTrue(videoStreams.isEmpty());
+        }
+    }
+
+    @Test
+    @Override
+    public void testAudioStreams() throws Exception {
+        final List<AudioStream> audioStreams = extractor().getAudioStreams();
+        assertNotNull(audioStreams);
+
+        if (expectedHasAudioStreams()) {
+            assertFalse(audioStreams.isEmpty());
+
+            for (final AudioStream stream : audioStreams) {
+                assertIsSecureUrl(stream.getUrl());
+
+                final int formatId = stream.getFormatId();
+                assertTrue("format id does not fit an audio stream: " + formatId,
+                        0x100 <= formatId && formatId < 0x1000);
```

Comment:
Could you explain why 0x100 <= formatId < 0x1000 and also above (video streams) or below (subtitles) and add comments for it?

Reply:
Ok